fixtures/proc-macro/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::sync::Arc;
+
+#[derive(uniffi::Record)]
+pub struct One {
+    inner: i32,
+}
+
+#[derive(uniffi::Record)]
+pub struct Two {
+    a: String,
+    b: Option<Vec<bool>>,
+}
+
+#[derive(uniffi::Record)]
+pub struct Three {
+    obj: Arc<Object>,
+}
+
+#[derive(uniffi::Object)]
+pub struct Object;
+
+#[uniffi::export]
+fn make_one(inner: i32) -> One {
+    One { inner }
+}
+
+#[uniffi::export]
+fn take_two(two: Two) -> String {
+    two.a
+}
+
+#[uniffi::export]
+fn make_object() -> Arc<Object> {
+    Arc::new(Object)
+}
```

Comment:
It's been nice seeing this feature grow.  There's a quite a bit of functionality here and it's expressed in a simple and clean way.